test/helpers/ProposalTest.sol

```diff
@@ -0,0 +1,336 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.20;
+
+import {ICompoundTimelock} from
+  "@openzeppelin/contracts/governance/extensions/GovernorTimelockCompound.sol";
+import {IERC20} from "@openzeppelin/contracts/interfaces/IERC20.sol";
+import {IGovernor} from "@openzeppelin/contracts/governance/IGovernor.sol";
+
+import {TestableProposeScript} from "test/helpers/TestableProposeScript.sol";
+import {IGovernorAlpha} from "src/interfaces/IGovernorAlpha.sol";
+import {RadworksGovernorTest} from "test/helpers/RadworksGovernorTest.sol";
+
+abstract contract ProposalTest is RadworksGovernorTest {
+  //----------------- State and Setup ----------- //
+
+  IGovernorAlpha governorAlpha = IGovernorAlpha(GOVERNOR_ALPHA);
+  ICompoundTimelock timelock = ICompoundTimelock(payable(TIMELOCK));
+  uint256 initialProposalCount;
+  uint256 upgradeProposalId;
+
+  // As defined in the GovernorAlpha ProposalState Enum
+  uint8 constant PENDING = 0;
+  uint8 constant ACTIVE = 1;
+  uint8 constant DEFEATED = 3;
+  uint8 constant SUCCEEDED = 4;
+  uint8 constant QUEUED = 5;
+  uint8 constant EXECUTED = 7;
+
+  // From GovernorCountingSimple
+  uint8 constant AGAINST = 0;
+  uint8 constant FOR = 1;
+  uint8 constant ABSTAIN = 2;
+
+  function setUp() public virtual override {
+    RadworksGovernorTest.setUp();
+
+    if (_useDeployedGovernorBravo()) {
+      // Use the actual live proposal data when it gets put on chain
+      upgradeProposalId = 17; // assume this is the next proposal
+      // Since the proposal was already submitted, the count before its submissions is one less
+      initialProposalCount = governorAlpha.proposalCount() - 1;
+    } else {
+      initialProposalCount = governorAlpha.proposalCount();
+
+      TestableProposeScript _proposeScript = new TestableProposeScript();
+      // We override the deployer to use an alternate delegate, because in this context,
+      // lonser.eth already has a live proposal
+      _proposeScript.overrideProposerForTests(0x69dceee155C31eA0c8354F90BDD65C12FaF5A00a);
+
+      upgradeProposalId = _proposeScript.run(governorBravo);
+    }
+  }
+
+  //--------------- HELPERS ---------------//
+
+  // a cheat for fuzzing addresses that are payable only
+  // Why is this no longer in standard cheats? JJF
+  // see https://github.com/foundry-rs/foundry/issues/3631
+  function assumePayable(address addr) internal virtual {
+    (bool success,) = payable(addr).call{value: 0}("");
+    vm.assume(success);
+  }
+
+  function _assumeReceiver(address _receiver) internal {
```

Comment:
love the comments in this section. it may be more explicit to name this method `_assumeValidReceiver`, but current works too

Reply:
This was the name used in the PoolTogether code.. keeping for consistency with that.